File: Sem1-2/CPP_Java2024/List3/CPP/src/Hexagon.cpp
#include <Hexagon.hpp>
#include <utils.hpp>

#include <stdexcept>
#include <cmath>

Hexagon::Hexagon(double side) : Shape("Hexagon"), side(side)
{
    if(side < 0)
    {
        throw std::invalid_argument(LOG_LOC() + "Hexagon side cannot be negative, got: " + std::to_string(side));
    }
}

double Hexagon::Circumference() const
{
    return 6.0 * side;
}

double Hexagon::Area() const
{
    return 1.5 * std::sqrt(3) * side * side;
}

File: Sem1-2/CPP_Java2024/List3/JAVA/Hexagon.java
public class Hexagon extends Shape
{
    protected final double side;

    public Hexagon(double side) throws IllegalArgumentException
    {
        super("Hexagon");
        if(side < 0)
        {
            throw new IllegalArgumentException("Hexagon side cannot be negative, got: " + side);
        }
        this.side = side;
    }

    @Override
    public double Circumference()
    {
        return 6.0 * side;
    }

    @Override
    public double Area()
    {
        return 1.5 * Math.sqrt(3) * side * side;
    }
}
